chamber/storages/boto3.py

```diff
@@ -30,36 +31,42 @@ def force_bytes_content(content, blocksize=1024):
     return content, False
 
 
-def s3_presigned_download_url(url=None, file=None, expiration=3600):
-    if not url and not file:
-        raise ValueError('You must provide either a "url" or "file".')
+def get_storage_class(s3_storage_class, default_storage_class=DefaultStorage):
+    return storage_class if settings.AWS_S3_ON else default_storage_class
 
-    if url:
-        return url
-    if not settings.AWS_S3_ON:
-        return file.url
 
-    s3 = boto3.client('s3', config=Config(region_name=settings.AWS_REGION, signature_version='s3v4'))
-    url = s3.generate_presigned_url(
-        'get_object',
-        Params={
-            'Bucket': file.storage.bucket_name,
-            'Key': file.storage.get_full_relative_path(file.name),
-        },
-        ExpiresIn=expiration,
-    )
-    return url
+def get_storage_instance(s3_storage_class, default_storage_class=DefaultStorage):
+    return s3_storage_class() if settings.AWS_S3_ON else default_storage_class()
 
 
-def get_storage_instance(storage_class, default_storage_class=DefaultStorage):
-    return storage_class() if settings.AWS_S3_ON else default_storage_class()
+class BaseS3Storage(S3Boto3Storage):
 
+    def save(self, name, content, max_length=None):
+        content, _ = force_bytes_content(content)
+        return super().save(name, content, max_length)
 
-class S3Storage(S3Boto3Storage):
+
+class BasePrivateS3Storage(BaseS3Storage):
+
+    expiration = 3600
 
     def get_full_relative_path(self, filename):
         return '{}/{}'.format(self.location, filename) if self.location else filename
 
-    def save(self, name, content, max_length=None):
-        content, _ = force_bytes_content(content)
-        return super().save(name, content, max_length)
+    def url(self, name):
```

Comment:
V předkovi se dělá toto:
`name = self._normalize_name(self._clean_name(name))`

V `_normalize_name` se dělá `safe_join(self.location, name)`, tedy stejná věc jako v `get_full_relative_path`. Tedy vyhodil bych `get_full_relative_path` a použil `_normalize_name`

Reply:
Je to interní API django-storages. Přijde mi lepší zrovna tuhle drobnost mít u sebe, než se spoléhat, že to v dalších verzích nezmění.

Reply:
my ale vycházíme z nějaké verze boto3 takže pokud už to bylo v předkovy tak není nutné to dělat znovu.. Díky Jirko do předka jsem nekoukal. BTW překavpuje mě, že tu URL neumí generovat nějak jednodušeji